tests/test.py

```diff
@@ -320,16 +316,49 @@ def run_terraform(*, image: str, final: bool = False):
             expected_number_of_logs = number_of_security_tools
         test_log_length = f"if [[ $(wc -l /var/log/easy_infra.log | awk '{{print $1}}') != {expected_number_of_logs} ]]; then exit 1; fi"
         command = f'/bin/bash -c "terraform init -backend=false && {test_log_length}"'
-        autodetect_environment["AUTODETECT"] = autodetect_status
+        learning_mode_and_autodetect_environment["AUTODETECT"] = autodetect_status
         utils.opinionated_docker_run(
             image=image,
             volumes=terraform_autodetect_volumes,
             command=command,
-            environment=autodetect_environment,
+            environment=learning_mode_and_autodetect_environment,
             expected_exit=0,
         )
         num_tests_ran += 1
 
+    # Ensure autodetect finds the appropriate terraform configs, which can be inferred by the number of logs written to /var/log/easy_infra.log
+    #
+    # This test ensure that, when DISABLE_SECURITY is true, the provided command is still run for each of the testing sub-directories.  It will exit
+    # non-zero on the first instance of a failed command, which should occur only when it encounters an invalid configuration.
+    test_terraform_dir = tests_test_dir.joinpath("terraform")
+    # There is always one log for each security tool, regardless of if that tool is installed in the image being used.  If a tool is not in the PATH
+    # and executable, a log message indicating that is generated.
+    LOG.debug("Testing autodetect mode with DISABLE_SECURITY on")
+    disable_security_environment = copy.deepcopy(environment)
+    disable_security_environment["DISABLE_SECURITY"] = "true"
+    disable_security_and_autodetect_environment = copy.deepcopy(
+        disable_security_environment
+    )
+    for autodetect_status in ["true", "false"]:
+        if autodetect_status == "true":
+            # number_of_* vars come from the above LEARNING_MODE test
+            expected_number_of_logs = (
+                number_of_security_tools * number_of_testing_folders
+            )  # TODO: Need to account for the order of the globbing here
```

Comment:
?

Reply:
Ah it's because `terraform` and my command will both exit 1... Need to make those different to get the test error based on a difference in log entries.